cudaPackages.nsight_systems: fix autoPatchelf errors (and more)

Teofil98 opened the pull request:
## Things done

Add the following missing dependencies for cudaPackages.nsight_systems: boost178, e2fsprogs and libQtPropertyBrowser.so

Should address #354189 

There are still some issues, possibly with a wrapper somewhere.
Running `nix-build -A cudaPackages.nsight_systems` completes successfully. The `nsys` binary can then be run without problems from the `result` folder.

However, creating a simple shell with `cudaPackages.nsight_systems` and running `nsys` gives the following error:
```
Error: Nsight Systems #VERSION_RSPLIT# hasn't been installed with CUDA Toolkit #CUDA_MAJOR#.#CUDA_MINOR#
```

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).



---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mcwitt committed:
cudaPackages.nsight_systems: fix binary discovery

Fixes the "#VERSION_RSPLIT#" error

(cherry picked from commit 88b047562543dad746c8fb3992193e859f99f7b5)

Commit:
cudaPackages_12.nsight_systems: fix autoPatchelf errors

Adds a patch-version of boost hard-coded for a particular release

SomeoneSerge commented:
> `Error: Nsight Systems #VERSION_RSPLIT# hasn't been installed with CUDA Toolkit #CUDA_MAJOR#.#CUDA_MINOR#`


https://github.com/NixOS/nixpkgs/issues/232458

SomeoneSerge committed:
cudaPackages.nsight_systems: fix rmPatterns

Quotation prevented the deletion of the vendored libssl, leading to runtime errors

SomeoneSerge commented:
Pushed again to recover the commit metadata. I can merge if somebody confirms the gui works

Snektron commented:
Seems there is still something missing. I'm getting this error while trying to run nsys-ui:
```
$ QT_QPA_PLATFORM=xcb result-bin/bin/nsys-ui
                                                                                                                                                                   zooi-env 
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

################# ERROR: CrashReporter #################
CrashReporter

Qt initialization failed

Could not find the Qt platform plugin "xcb" in ""
 - Qt plugin search path: /nix/store/bnvaxw6d2dgj1rxva6pq3g4zxkmyf0f0-nsight_systems-2023.4.4.54-bin/nsight-systems/2023.4.4/host-linux-x64
 - Qt plugin search path: /nix/store/bnvaxw6d2dgj1rxva6pq3g4zxkmyf0f0-nsight_systems-2023.4.4.54-bin/nsight-systems/2023.4.4/host-linux-x64
Application is going to abort
sh: line 1: xmessage: command not found
/nix/store/bnvaxw6d2dgj1rxva6pq3g4zxkmyf0f0-nsight_systems-2023.4.4.54-bin/nsight-systems/2023.4.4/host-linux-x64/nsys-ui: line 63: 158135 Aborted                 (core dumped) "$NV_AGORA_PATH/CrashReporter" --hide-stack "NVIDIA Nsight Systems" "NsightSystems" "2023.4.4 (Build 2023.4.4.54-234433681190v0)" "$NV_QUADD_PATH" "$@"
```

gou4shi1 commented:
https://github.com/NixOS/nixpkgs/pull/355470#issuecomment-2476181650
same for me

aidalgol commented on pkgs/development/cuda-modules/cuda/overrides.nix:
```diff
@@ -328,8 +331,11 @@ filterAndCreateOverrides {
-            rm -r "$path"
+            rm -r $path
           done
+        ''
+        + ''
+          patchShebangs nsight-systems
```
Am I missing something, or could this just be combined into one multi-line string instead of concatenating two multi-line strings?

samuela commented on pkgs/development/cuda-modules/cuda/overrides.nix:
```diff
@@ -328,8 +331,11 @@ filterAndCreateOverrides {
         + ''
           for path in $rmPatterns; do
-            rm -r "$path"
+            rm -r $path
```
why get rid of double quotes? shouldn't they be included to prevent https://www.shellcheck.net/wiki/SC2086?

prusnak commented:
superseded by https://github.com/NixOS/nixpkgs/pull/393451

please open a new PR if there are some outstanding issues that need to be fixed
